### DISCUSSION TITLE
Explanation of "Optimized Lambdas" for Next.JS

### DISCUSSION THREAD

**Discussion by cryptiklemur** [vercel/vercel#5135] [Help] [answered]
In regards to this page: https://github.com/vercel/vercel/blob/master/errors/next-legacy-routes-optimized-lambdas.md

I have specified in my `vercel.json` a function config that gives my `src/pages/api/**` routes a high `maxDuration`, so i get the error that I'm not using the optimized lambdas. Is there any information on what this means? What am I losing here?

Is there a way to get the optimized lambda's while still maintaining my maxDuration on api routes?

**Comment by felipebrahm** [comment]
@aequasi, where you able to find more information about this? I need to increase the memory for just one of my API routes and now I'm seeing this warning. I'm wondering if I'm losing this optimization for _all_ my routes or just the one included in my config file. The message seems ambiguous.

**Comment by mcsdevv** [comment, marked as answer]
Hey @aequasi @felipebrahm, if you use `routes` currently, you will opt out of this on **all** Serverless Functions. However, we are working on improving this to allow for use with `routes` and `functions` configuration properties too.

**Comment by ArrayKnight** [comment]
@mcsdevv Is there any sort of roadmap / timeline on when we could expect NextJS API routes to be able to use optimized lambdas?

https://github.com/vercel/vercel/blob/master/errors/next-functions-config-optimized-lambdas.md

I don't necessarily need to set any of the options, just want to have my preview route (for my headless CMS) as performant as possible.

**Reply by mcsdevv** [reply]
@ArrayKnight Next.js API Routes are already part of this, they are only opted out if you use the `routes` or `functions` configuration properties in the `vercel.json` file.

**Reply by ArrayKnight** [reply]
If I don't include a `vercel.json` with:
```json
{
    "functions": {
        "src/pages/api/**": {
            "memory": 1024
        }
    }
}
```

Then my preview route `src/pages/api/preview` is not accessible.

If I do include this file I get the opt-out message, but my preview route is accessible.

Also, I would like to be able to remove the memory property (and just use the default) but Vercel requires at least one property be set.

@mcsdevv Is there some other way of accomplishing this that I haven't found?
